### DISCUSSION TITLE
[Prettier Plugin] - Utility sorting order involving directions

### DISCUSSION THREAD

**Discussion by lutaok** [tailwindlabs/tailwindcss#14068] [Help]
Hello everyone!
I'd like to get some help on the subject's matter.

I'm noticing that when it comes to utility classes like `mb-2` or `mt-2` the order seems a bit off.
From what I could grasp from the configuration, I should expect `text-sm mb-2 mt-2` to be ordered `mt-2 mb-2 text-sm`.
The prettier plugin, instead, orders them as `mb-2 mt-2 text-sm`, is this wanted?

I can replicate this behaviour on the Tailwind Playground, which I would assume should start with ordered classes: https://play.tailwindcss.com/
If I save without touching anything, I see that classes with "direction" swap places.

I'll provide another example class string:
`ml-2 my-4 mx-2 mr-4 m-4 mt-12 mb-4`

Based on the configuration I would expect this order:
`m-4 mx-2 my-4 mt-12 mr-4 mb-4 ml-2`

But I get this, instead:
`m-4 mx-2 my-4 mb-4 ml-2 mr-4 mt-12`

I have done a search of previous discussions and found this one that had a very similar issue:
https://github.com/tailwindlabs/tailwindcss/discussions/13510
I'm using MacOS so probably not what's been discussed there.

Also, I just came across this discussion:
https://github.com/tailwindlabs/tailwindcss/discussions/13833
I'm continuing the already great work done on Biome's implementation of the Tailwind Prettier Plugin, so collaboration is well welcomed 😄 

Thanks in advance for any help you can give me, I'd like to know if there's a particular consideration being done around those kind of values when sorting 😃

**Comment by florence0239** [comment]
Hello,
 It sounds like you’re encountering some unexpected behavior with the Tailwind CSS Prettier plugin’s class sorting. Let’s break down what’s happening and why.

Tailwind CSS Class Sorting with Prettier
The Tailwind CSS Prettier plugin is designed to automatically sort your utility classes in a specific order. This order is based on the way Tailwind organizes its CSS, which aims to group related utilities together and ensure that classes that override others appear later in the list.

Expected vs. Actual Order
From your examples:

Expected Order: `m-4 mx-2 my-4 mt-12 mr-4 mb-4 ml-2`
Actual Order: `m-4 mx-2 my-4 mb-4 ml-2 mr-4 mt-12`
The plugin sorts classes in a way that might not always align with your expectations, especially when it comes to margin and padding utilities. The order is influenced by the box model, with high-impact layout classes appearing first and more decorative classes later.

Why the Difference?
The difference you’re seeing is likely due to the plugin’s sorting logic, which prioritizes certain classes over others. For example, margin and padding classes are sorted based on their impact on the layout, with `m-4` (all sides) coming before more specific classes like` mt-12` (top) and `mb-4` (bottom).

Tailwind Playground
The Tailwind Playground uses the same sorting logic as the Prettier plugin, so the behavior you’re seeing there is consistent with what the plugin does

Hope this will help you.
Best regards,
florence0239
[Official Site](https://www.health-insurancemarket.com)
